### PR TITLE
core: add table tests for unsigned data

### PR DESCRIPTION
**Pull request by dB2510**
Adds table tests for unsigned data implementations.

category: test
ticket: none

### CODE REVIEW

File: core/unsigneddata_test.go [hunk 1]
```diff
@@ -24,30 +24,41 @@ import (
...
-	clone, err := block.Clone()
-	require.NoError(t, err)
-	block2 := clone.(core.VersionedBlindedBeaconBlock)
-	slot2, err := block2.Slot()
-	require.NoError(t, err)
+			b2, err := clone.MarshalJSON()
+			require.NoError(t, err)
 
-	require.Equal(t, slot1, slot2)
+			require.Equal(t, b1, b2)
```
Comment (xenowits) [inline core/unsigneddata_test.go:61]: Let's also add another line of test for `UnmarshalJSON()`. Like, for VersionedBeaconBlock:
```go
	x := new(core.VersionedBeaconBlock)
	require.NoError(t, x.UnmarshalJSON(b1))
	require.Equal(t, *x, test.data)
```

**Review by xenowits — APPROVED** [review]

File: core/unsigneddata_test.go (hunk 1 above)
Comment (dB2510) [reply core/unsigneddata_test.go:61]: i don't think we need this since Clone() method already takes care of this

Comment (xenowits) [reply core/unsigneddata_test.go:61]: Btw, Clone() also uses the `MarshalJSON()` method, so actually, we don't need to test the marshal/unmarshal methods explicitly.

**Review by corverroos — APPROVED** [review]

File: core/unsigneddata_test.go (hunk 1 above)
Comment (dB2510) [reply core/unsigneddata_test.go]: sure

**Pull request merged**